File: solution/169.majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
  let count = 1,
    temp = 0;
  for (let i = 1; i < nums.length; i++) {
    if (nums[temp] == nums[i]) {
      count++;
    } else {
      count--;
      if (count == 0) {
        temp = i;
        count = 1;
      }
    }
  }
  return nums[temp];
};

File: solution/169.majority-element/majority-element.c
int majorityElement(int* nums, int numsSize) {
    int i;
    int count = 1;
    int temp = 0;
    for(i=1;i<numsSize;i++)
    {
        if(nums[temp] == nums[i])
            count++;
        else
        {
            count--;
            if(count == 0)
            {
                temp = i;
                count =1;
            }
        }
    }
    return nums[temp];
}